### DISCUSSION TITLE
problem with the receiver's location

### DISCUSSION THREAD

**Discussion by ayaywzn** [SPECFEM/specfem3d#1624] [SPECFEM3D General]
Hi all,
I am new to SPECFEM3D. I have a problem with the receiver's location and your help is appreciated. I am running the specfem3D_Cartesian. The SEM model considers the actual topography of the simulation region. I specify several receivers at the surface with depth of 0 m. However, the target receivers are inconsistent with my preconceptions original coordinates I specified, i.e.:

station # 2 N N3308N
original latitude: 3260730.
original longitude: 230386.0
original x: 230386.0
original y: 3260730.
original depth: 0.0000000E+00 m
horizontal distance: 43.3977160057072
target x, y, z: 230386.0 3260730. 1187.291
closest estimate found: 11.62000 m away
receiver located in slice 159
in element 11250
in elastic domain
at coordinates:
xi = 0.624910007199495
eta = 1.00000000000000
gamma = 1.00000000000000
rotation matrix:
nu1 = 1.00000000000000 0.000000000000000E+000
0.000000000000000E+000
nu2 = 0.000000000000000E+000 1.00000000000000
0.000000000000000E+000
nu3 = 0.000000000000000E+000 0.000000000000000E+000
1.00000000000000
x: 230386.000000000
y: 3260722.00000000
depth: 8.42759222328186 m
z: 1178.86330133141

I do not know why the receivers are moved. I would to know how to solve this problem in order to get the results in the accurate position.
Thank you in advance,

**Comment by homnath** [comment]
Hello,

Is it possible to share the mesh and DATA file?

Best,
Hom Nath
________________________________
From: ayaywzn ***@***.***>
Sent: Wednesday, July 19, 2023 1:09 AM
To: SPECFEM/specfem3d ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [SPECFEM] problem with the receiver's location (Discussion #1624)


Hi all,
I am new to SPECFEM3D. I have a problem with the receiver's location and your help is appreciated. I am running the specfem3D_Cartesian. The SEM model considers the actual topography of the simulation region. I specify several receivers at the surface with depth of 0 m. However, the target receivers are inconsistent with my preconceptions original coordinates I specified, i.e.:

station # 2 N N3308N
original latitude: 3260730.
original longitude: 230386.0
original x: 230386.0
original y: 3260730.
original depth: 0.0000000E+00 m
horizontal distance: 43.3977160057072
target x, y, z: 230386.0 3260730. 1187.291
closest estimate found: 11.62000 m away
receiver located in slice 159
in element 11250
in elastic domain
at coordinates:
xi = 0.624910007199495
eta = 1.00000000000000
gamma = 1.00000000000000
rotation matrix:
nu1 = 1.00000000000000 0.000000000000000E+000
0.000000000000000E+000
nu2 = 0.000000000000000E+000 1.00000000000000
0.000000000000000E+000
nu3 = 0.000000000000000E+000 0.000000000000000E+000
1.00000000000000
x: 230386.000000000
y: 3260722.00000000
depth: 8.42759222328186 m
z: 1178.86330133141

I do not know why the receivers are moved. I would to know how to solve this problem in order to get the results in the accurate position.
Thank you in advance,

**Comment by ayaywzn** [comment]
Hi,


Thank you very much for your reply. I'm sorry I'm not at school at the moment. I will send you the data after I return to school in two days.


Thank you in advance,


Best,


Zhang.

**Comment by ayaywzn** [comment]
Hi,

Sorry to reply you so late.
Attached are the DATA and OUTPUT file. The local grid and station location are shown in Figure 1.
Thank you very much for your help.

Best,
Zhang
[Files.zip](https://github.com/SPECFEM/specfem3d/files/12190536/Files.zip)

**Comment by homnath** [comment]
Hi Zhang,

I tested your example with the latest development version. I get

maximum error in location of all the receivers:    7.541802      m

You have a complex topography and some of the stations are located in the steep topography regions. This seems to be the reason for the error.
 Although, the error is still less than
*** Min GLL point distance =    22.48438
*** Min element size =    130.2031
it is not accurate due to the severely distorted elements. Location is an interactive procedure. I tried increasing the number of iterations but it did not improve. Refining the mesh would improve the location but that may be costly.

Best,
Hom Nath

---------------------------------------
Hom Nath Gharti, PhD
Assistant Professor | Digital Earth Scientist
https://www.digitalearthscience.com/
Department of Geological Sciences and Geological Engineering
Miller Hall 314, 36 Union St
Queen’s University
Kingston, ON K7L 3N6, Canada
<https://digitalearthscience.com/>

Queen’s University is situated on traditional
Anishinaabe and Haudenosaunee Territory.<https://www.queensu.ca/encyclopedia/t/traditional-territories>
________________________________

**Reply by zmz20** [reply]
Hi,

Thank you very much for your kind help. I will try to refine the mesh. If it does not work, I think that output the result of all the integral points of the element, then program myself to do the post-processing interpolation, seems to be an optional method.

Thank you again.

Best,
Zhang

**Reply by zmz20** [reply]
Hi,

  Hope everything's OK with you. I am the author, this is my another account.
  It seems that the reason is not simply the steep topography, of course the question is related to steep topography, I think. 
  As shown in Figure 1, there are two target nodes. TargetNode1 is near the boundary of element. When “subroutine locate_point_in_mesh” is running, FindElement1 is regarded as the most suitable element which TargetNode1 is located in. And this is an error result. Therefore, FindNode1 is regarded as the final location of TargetNode1, and I get _“closest estimate found:    11.62000      m away       xi    =   0.624910007199495        **eta   =    1**        **gamma =    1**_”. 
  However, when it comes to TargetNode2, which is away from the element boundary, the right location is found by SPECFEM3D and I get “closest estimate found:  2.9103830E-11  m away”.
  I think “subroutine locate_point_in_mesh” should be improved. And it is the reason that increasing iteration number does not work, I think. I will try to improve it, but I am not sure if I can complete the work because I am new to SPECFEM3D. I would like to ask your opinion on this issue.

  Thank you very much.
![问题描述](https://github.com/SPECFEM/specfem3d/assets/73430593/bc3b1ffa-29b2-4ec5-ba9a-8193d2944a41)

Best,
Zhang

**Reply by homnath** [reply]
That is interesting. It may also be due to the tolerance we set for the location. Let's fix it. Please feel free to improve and make a pull request.

Best,
Hom Nath

**Reply by zmz20** [reply]
Thank you very much. I'm not entirely sure I've found the key problem. I'll test it again. Recently, I will try to improve it. If you solve this problem, I also hope to get your help. Let's keep in touch.

Best,
Zhang

**Reply by zmz20** [reply]
Hi,
First of all, I need to correct my guess in the previous reply. Maybe the problem is not in “subroutine locate_point_in_mesh”.
I test my model again. In the previous tests, the option USE_SOURCE_RECEIVERS_Z is off. And I found that the error "closest estimate found" does not mean that the found points is wrong. As shown in Figure, I want to get FindNode1 at the surface and 0 burial is given in the STATIONS file. However, the TargetNode1 that SPECFEM calculated is several meters above the surface. Then, the right FindNode1 can be found although I get "closest estimate found:  8 m". But sometimes, we may get the wrong result if the target node is closer to another element, just as I described in the previous reply.
Then, I turn on the option USE_SOURCE_RECEIVERS_Z and this issue can be improved to some extent. However, there are still a few points that will deviate from the target point. It is because the steep topography, resulting in severe element surface distortion. But this will not make a big impact to my results. Maybe there is something wrong with my idea.
Thank you very much for your help. And I will keep my eyes on this issue.
Best,
Zhang
![Figure2](https://github.com/SPECFEM/specfem3d/assets/73430593/69de36f9-d2af-4df3-b073-832f865aaa29)

**Comment by homnath** [comment]
Hi Zhang,

"USE_SOURCE_RECEIVERS_Z" should not affect the location accuracy. That just facilitates the alternative way to input the source-receiver Z coordinates.
If USE_SOURCE_RECEIVERS_Z is .TRUE., the program assumes the input depth as an actual Z coordinate in the mesh. If USE_SOURCE_RECEIVERS_Z is .FALSE. the actual Z coordinate is estimated from the depth, elevation and burrial. In this case, the depth is measured from Zero elevation surface positive downward. Both options should give you identical Z coordinates if properly used.

Best,
Hom Nath


---------------------------------------
Hom Nath Gharti, PhD
Assistant Professor | Digital Earth Scientist
https://www.digitalearthscience.com/
Department of Geological Sciences and Geological Engineering
Miller Hall 314, 36 Union St
Queen’s University
Kingston, ON K7L 3N6, Canada
<https://digitalearthscience.com/>

Queen’s University is situated on traditional
Anishinaabe and Haudenosaunee Territory.<https://www.queensu.ca/encyclopedia/t/traditional-territories>
________________________________
